src/AsyncAwsS3/AsyncAwsS3Adapter.php

```diff
@@ -285,7 +285,7 @@ public function copy(string $source, string $destination, Config $config): void
     {
         try {
             /** @var string $visibility */
-            $visibility = $this->visibility($source)->visibility();
+            $visibility = $config->get(Config::OPTION_VISIBILITY, $this->visibility($source)->visibility());
```

Comment:
The call to read the visibility can be omitted when the value is configured.

```suggestion
            $visibility = $config->get(Config::OPTION_VISIBILITY) ?: $this->visibility($source)->visibility();
```